### ISSUE TITLE
Error codes are not removed from extension health checks

### ISSUE THREAD

**Issue by timuthy** [gardener/gardener#5205]
**How to categorize this issue?**

/area control-plane
/kind bug
/priority 2

**What happened**:
A cluster which had a cloud quota issue was correctly assigned a `ERR_INFRA_QUOTA_EXCEEDED` code for the `EveryNodeReady` condition of the `worker` resource. However, after the quota was increased and the actual issue was gone the code was not removed:

```
  {
    "codes": [
      "ERR_INFRA_QUOTA_EXCEEDED"
    ],
    "lastTransitionTime": "2021-12-22T08:44:56Z",
    "lastUpdateTime": "2021-12-22T08:54:29Z",
    "message": "All health checks successful",
    "reason": "HealthCheckSuccessful",
    "status": "True",
    "type": "EveryNodeReady"
  },
```

**What you expected to happen**:
The error code to be removed from the condition.

**How to reproduce it (as minimally and precisely as possible)**:
- Use a shoot and provoke an error so that a [known code](https://github.com/gardener/gardener/blob/a5d3a55d12dfb76e530f0e07d56745fae204dba3/pkg/apis/core/types_common.go#L25-L49) is assigned to the worker's condition.
- Solve the issue that provoked the error.
- Verify that condition is healthy again but error code is still part of the condition.

**Environment**:

- Gardener version: `v1.37.2` 
- Kubernetes version (use `kubectl version`): 1.21.5
- Cloud provider or hardware configuration: GCP

**Comment by rfranzke** [comment]
Seems similar to https://github.com/gardener/gardener/pull/4885. Do we know the version of the extension controller which was adding this error code? Does it already contain this fix?

**Comment by timuthy** [comment]
Good point, thanks for posting this PR. The fix was delivered with `gardener/gardener v1.34.0` but the released GCP extension only has a vendored version of `v1.32.0`.
/close

**Comment by rfranzke** [comment]
Thanks for checking!
